torch/nn/functional.py

```diff
@@ -1425,7 +1431,7 @@ def embedding(input, weight, padding_idx=None, max_norm=None, norm_type=2.,
         # with torch.no_grad():
         #   torch.nembedding_renorm_
         # remove once script supports set_grad_enabled
-        torch.no_grad_embedding_renorm_(weight, input, max_norm, norm_type)
+        no_grad_embedding_renorm_(weight, input, max_norm, norm_type)
```

Comment:
You can't make embedding into an aten op because you can't set NoGrad within aten. The entire function could be made into a native op. However, we can't run optimizations within native functions, so it's preferable to have as little logic as possible within them. As is, If we know max_norm is None at compile time, then we will remove the if branch that calls no_grad_embedding_renorm_. If the whole op was native then we couldn't optimize it.

Reply:
This gets into the bigger question of  "is moving an operation to aten now slower because we can't run the JIT on it" but let's not worry about that for now. I see, I forgot that NoGrad isn't a part of aten, thanks for that clarification